community/contributing/cpp_usage_guidelines.rst

```diff
@@ -0,0 +1,96 @@
+.. _doc_cpp_usage_guidelines:
+
+C++ usage guidelines
+====================
+
+Rationale
+---------
+
+Since Godot 4.0, the C++ standard used throughout the codebase is a subset of
+**C++17**. While modern C++ brings a lot of opportunities to write faster, more
+readable code, we chose to restrict our usage of C++ to a subset for a few
+reasons:
+
+- It makes it easier to review code in online editors. This is because engine
+  contributors don't always have access to a full-featured IDE while reviewing
+  code.
+- It makes the code easier to grasp for beginner contributors (who may not be
+  professional C++ programmers). Godot's codebase is known to be easy to learn
+  from, and we'd like to keep it that way :)
```

Comment:
While both are true, I don't think they're the only reasons why we use a subset and not full-blown over-engineered C++ (hint in the qualifier). This should likely be discussed further with core devs to come with a strong rationale, otherwise we'll get criticism for taking arbitrary technical decisions just for the sake of non-IDE reviews and junior devs.

Reply:
Also, the smiley should probably be dropped. It feels kinda condescending.